int red
int amber
int green
int blue
int white
int x
int y
---
string cmdstr